src/OpenTelemetry/Metrics/HistogramBuckets.cs

```diff
@@ -104,5 +186,18 @@ public bool MoveNext()
                 return false;
             }
         }
+
+        private class Bucket
```

Comment:
@mic-max Would you add `sealed` here? Not sure if it will make a big impact but I try to seal everything I can. Based off of what I have seen being done in runtime: https://github.com/dotnet/runtime/pull/49969